explainer.md

```diff
@@ -186,225 +181,236 @@ https://user-images.githubusercontent.com/93594/141100217-ba1fa157-cd79-4a9d-b3b
+
+### Via JavaScript
+
+The JavaScript API extends on the capabilities of the CSS API.
 
-- [Open question](https://github.com/WICG/shared-element-transitions/issues/86) What's the deadline for calling `performTransition`?
+```js
+document.addEventListener("beforepageshow", async (event) => {
+  if (!event.transition) return;
+  event.transition.setElement(document.documentElement, "root");
+  event.transition.setElement(document.querySelector(".header"), "header");
 
-Note that using pseudo-elements here implies that the developer can not change the DOM structure of transition elements. An alternate approach considered to provide this extensibility uses shadow DOM. The pseudo-element option was preferred to make it easier to build on concepts which already exist in the platform and ensure ease of implementation. See [issue](https://github.com/WICG/shared-element-transitions/issues/93) for detailed discussion.
+  await event.transition.ready;
 
-- [Open question](https://github.com/WICG/shared-element-transitions/issues/87): Is the freedom above a feature or a bug?
+  // The pseudo-elements are now accessible and can be animated:
+  document.documentElement.animate(keyframes, {
+    ...animationOptions,
+    pseudoElement: "::page-transition(header)",
+  });
+});
+```
 
-## Signaling the end of a page transition
+`beforepageshow` is a new event which fires just before the page is shown. Unfortunately `pageshow` fires a long after the page is shown, after `window.onload`. The developer will have to add a listener to this somewhere that's executed before the page is shown.
 
-At the start of the transition, the browser could gather all the `Animation`s active on the stage, and assume the animation is complete once all animations finish.
+- `event.previousURL` - URL of the previous page. Is `null` if cross-origin, or if there was no previous page.
+- `event.direction` - `forward` if moving forward in history, `back` is moving back in history, `new` otherwise.
+- `event.transition` - `null` if Page-A did not offer any elements.
+- `event.transition.data` - Data set by Page-A via `setData`.
+- `event.transition.abandon()` - Abandon the transition.
+- `event.transition.setElement(element, tag, options)` - Same API as in Page-A.
+- `event.transition.ignoreCSSTaggedElements()` - Same API as in Page-A.
 
-In addition, the JS API could include a way to override this by letting the developer provide a promise which keeps the transition active, allowing for animations driven some other way, such as `requestAnimationFrame`.
+In future, this API could include:
 
-This is discussed in [#64](https://github.com/WICG/shared-element-transitions/issues/64).
+- A way to delay the start of the transition, eg to wait for an image to load.
+- A way to enumerate the pseudo elements, which will depend on [CSSPseudoElement](https://drafts.csswg.org/css-pseudo-4/#CSSPseudoElement-interface).
+- A way to manually associate Page-A and Page-B elements, to change the default.
```

Comment:
And this too. The associating of elements can already be controlled by the developer via tag names?